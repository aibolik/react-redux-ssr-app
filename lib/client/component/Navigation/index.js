import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import './index.scss';
import User from '../User';

const Navigation = props => React.createElement(
  'nav',
  { className: 'navigation' },
  React.createElement(
    'ul',
    { className: 'links' },
    React.createElement(
      'li',
      null,
      React.createElement(
        NavLink,
        { exact: true, to: '/', className: 'links__item', activeClassName: 'links__item--active' },
        'Home'
      )
    ),
    React.createElement(
      'li',
      null,
      React.createElement(
        NavLink,
        { to: '/feed', className: 'links__item', activeClassName: 'links__item--active' },
        'Feed'
      )
    )
  ),
  React.createElement(
    'div',
    { className: 'profile' },
    props.user ? React.createElement(User, props.user) : React.createElement(
      Link,
      { to: '/signup', className: 'links__item' },
      'Sign in'
    )
  )
);

export default Navigation;