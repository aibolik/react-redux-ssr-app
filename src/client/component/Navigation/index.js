import React from 'react';
import { NavLink } from 'react-router-dom';
import './index.scss';
import User from '../User';

const Navigation = props => (
  <nav className='navigation'>
    <ul className='links'>
      <li><NavLink exact to="/" className='links__item' activeClassName='links__item--active'>Home</NavLink></li>
      <li><NavLink to="/feed" className='links__item' activeClassName='links__item--active'>Feed</NavLink></li>
    </ul>
    <div className='profile'>
      <User name='Aibol' />
    </div>
  </nav>
);

export default Navigation;
