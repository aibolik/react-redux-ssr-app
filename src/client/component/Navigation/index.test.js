import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import Navigation from './index';
import renderer, { shollow as ShallowRenderer } from 'react-test-renderer';
import TestUtils from 'react-dom/test-utils';

describe('Test Navigation component', () => {
  const props = {
    user: {
      id: '5aa0dedae3fa01461ca2de06',
      name: 'Test Name'
    },
    logOut: jest.fn(() => {
      console.log('I am called')
    })
  };

  it('renders correctly without User props', () => {
    const tree = renderer.create(<BrowserRouter><Navigation /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with User props', () => {
    const tree = renderer.create(<BrowserRouter><Navigation {...props} /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
