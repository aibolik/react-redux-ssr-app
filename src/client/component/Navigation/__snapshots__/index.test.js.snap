// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Navigation component renders correctly with User props 1`] = `
<nav
  className="navigation"
>
  <ul
    className="links"
  >
    <li>
      <a
        aria-current={false}
        className="links__item"
        href="/"
        onClick={[Function]}
        style={undefined}
      >
        Home
      </a>
    </li>
    <li>
      <a
        aria-current={false}
        className="links__item"
        href="/feed"
        onClick={[Function]}
        style={undefined}
      >
        Feed
      </a>
    </li>
  </ul>
  <div
    className="profile"
  >
    <div
      className="user"
    >
      <div
        className="user__name"
      >
        Test Name
      </div>
      <div
        className="user__log-out"
        onClick={[MockFunction]}
      >
        Log out
      </div>
    </div>
  </div>
</nav>
`;

exports[`Test Navigation component renders correctly without User props 1`] = `
<nav
  className="navigation"
>
  <ul
    className="links"
  >
    <li>
      <a
        aria-current={false}
        className="links__item"
        href="/"
        onClick={[Function]}
        style={undefined}
      >
        Home
      </a>
    </li>
    <li>
      <a
        aria-current={false}
        className="links__item"
        href="/feed"
        onClick={[Function]}
        style={undefined}
      >
        Feed
      </a>
    </li>
  </ul>
  <div
    className="profile"
  >
    <a
      className="links__item"
      href="/signup"
      onClick={[Function]}
    >
      Sign Up
    </a>
    <a
      className="links__item"
      href="/signin"
      onClick={[Function]}
    >
      Sign In
    </a>
  </div>
</nav>
`;
